library/src/main/java/com/google/android/material/motion/runtime/Scheduler.java

```diff
@@ -172,6 +172,38 @@ public void addPlan(Plan plan, Object target) {
     getTargetScope(target).commitPlan(planInfo);
   }
 
+  /**
+   * Adds a {@link NamedPlan} to this scheduler.
+   * @param plan the {@link NamedPlan} to add to the scheduler.
+   * @param name the name by which this plan can be identified.
+   * @param target the target on which the plan will operate.
+   */
+  public void addNamedPlan(NamedPlan plan, String name, Object target) {
+    if (name == null || name.length() == 0) {
+      throw new AssertionError("A NamedPlan must have a name with more than zero characters");
+    }
+    PlanInfo planInfo = new PlanInfo();
+    planInfo.name = name;
+    planInfo.target = target;
+    planInfo.plan = plan.clone();
+    getTargetScope(target).commitAddNamedPlan(planInfo);
+  }
+
+  /**
+   * Removes a {@link NamedPlan} from this scheduler.
+   * @param name the name by which the named plan can be identified.
+   * @param target the target on which the named plan was added.
+   */
+  public void removePlanNamed(String name, Object target) {
+    if (name == null || name.length() == 0) {
+      throw new AssertionError("A NamedPlan must have a name with more than zero characters");
+    }
+    PlanInfo planInfo = new PlanInfo();
+    planInfo.name = name;
+    planInfo.target = target;
+    getTargetScope(target).commitRemoveNamedPlan(planInfo);
```

Comment:
hm I don't like (mis?)using a partial PlanInfo to just hold the name like that. I would rather see the api be `commitRemoveNamedPlan(name)`